packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -1720,8 +1720,25 @@ export class ApexDebug extends LoggingDebugSession {
 
       // log to console and notify client
       this.logEvent(message);
+      let reason = '';
+
+      // if breakpointid is found in exception breakpoint cache
+      // set the reason for stopped event to that reason
+      if (message.sobject.BreakpointId) {
+        const cache: Map<
+          string,
+          string
+        > = this.myBreakpointService.getExceptionBreakpointCache();
+        cache.forEach((value, key) => {
+          if (value === message.sobject.BreakpointId) {
+            if (key.indexOf('Exception') !== -1) {
+              reason = key.substring(key.lastIndexOf('/') + 1);
```

Comment:
What does this look like when the org is namespaced? 

How about an inner exception?
`public class MyClass {
  public class MyInnerException extends Exception {}
}`

Reply:
The listing does not provide namespaces for the inner exceptions. I created two classes MyClass and MyClass2 both with MyInnerException classes and the list shows the following:
![image](https://user-images.githubusercontent.com/4503716/34497798-dd0f4e56-efb3-11e7-9a9e-28245017b3ac.png)

Reply:
There's a work item for the apex-jorje bug shown in the screenshot. The work item suggest apex-jorje should label it MyClass.MyInnerException (instead of just MyInnerException) because that's what the user would have to type if they were writing Apex code using an inner class. That bug shouldn't affect what you're doing here. The breakpoint cache is keyed by typeref, so for the case of an inner class, the typeref would be MyClass$MyInnerException. My suspicion is the visual indicator shows "Paused on MyClass$MyInnerException". If that's true, could you update the reason to use a period instead of a dollar sign?

Reply:
Had a lot of issues getting this set up locally but finally able to confirm - it does show up as you suspected with "Paused on MyClass$MyInnerException". I'll replace the dollar sign with a period.

For a namespaced org, the type ref will be "namespace/MyClass$MyInnerException" but we will only show the exception name ("MyClass$MyInnerException). What should the behavior be for this? Should we show the full type ref or just the exception? Dollar signs will be changed to periods.

Reply:
@DatGuyJonathan here's what I have them looking like right now.
![image](https://user-images.githubusercontent.com/4503716/34506615-53efcda0-efe3-11e7-870b-3b265eb5348b.png)

![image](https://user-images.githubusercontent.com/4503716/34506617-5acea308-efe3-11e7-8d50-dac8cf04f8e2.png)

Reply:
We should show namespace.MyClass.MyInnerException. The use case would be an ISV debugging a namespaced subscriber org. The ISV's package would be namespaced so we should make it clear which namespace the exception class belongs to.

I suspect you're using `lastIndexOf('/')` because System exceptions are prefixed with `com/salesforce/api/exception/`. The alternative would be to just remove `com/salesforce/api/exception/` so then you can replace the remaining `/` with a period.

I believe you can also have an exception class inside a trigger. The typeref would be __sfdc_trigger/MyTrigger$MyException, so you'd have to remove `__sfdc_trigger/` as well.